src/lib/drivers/device/CMakeLists.txt

```diff
@@ -53,8 +53,13 @@ px4_add_library(drivers__device
 	${SRCS_PLATFORM}
 	)
 
+# px4_spibus_initialize (stm32_spibus_initialize)
 if (${PX4_PLATFORM} STREQUAL "nuttx")
-	target_link_libraries(drivers__device PRIVATE nuttx_arch)
+	if (NOT DEFINED CONFIG_BUILD_FLAT)
+	  target_link_libraries(drivers__device PRIVATE nuttx_karch)
+	else()
+	  target_link_libraries(drivers__device PRIVATE nuttx_arch)
+	endif()
 endif()
 
-target_link_libraries(drivers__device PRIVATE cdev)
+target_link_libraries(drivers__device PRIVATE px4_work_queue)
```

Comment:
This one surprises me.

Reply:
This looks very weird to me as well. I will check and revert this line if it is wrong - it certainly does look wrong, I don't remember why or where this came from...

Reply:
This was indeed some fumble in re-basing commits. I reverted this line separately in #19092 . Other changes in the same file were as intended